### PR TITLE
Comparison constraint usage analyzer

### LINKED ISSUE #329: Analyzer to warn against comparison operators in actual expression
Add an analyzer or set of analyzers to warn against using comparison operators (`==`, `!=`, `>`, `>=`, `<`, `<=`) in the actual expression of `Assert.That`
For example, `Assert.That(value > 0, Is.True)` is bad, it should be `Assert.That(value, Is.GreaterThan(0))`

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #329

### CODE REVIEW

**Review by mikkelbu — COMMENTED** [review]
I think this in an improvement and I've only found a couple of the smallest nitpick :). But I was wondering how difficult it would be to handle the expected / actual parts even better. Currently, if we have 
```csharp
var x = 20;
Assert.That(14 < x);
```
then it is fixed into 
```csharp
var x = 20;
Assert.That(14, Is.LessThan(x));
```
which is correct, but with the expected and the actual value swapped. I would prefer 
```csharp
var x = 20;
Assert.That(x, Is.GreaterThan(14));
```
So if one of the sides of the binary operator is a constant then we know that this side is the expected value (and then this should be placed inside the expected expression. If neither side is a constant, then I think we could keep the current translation as fallback.

File: src/nunit.analyzers.sln [hunk 1]
```diff
@@ -81,6 +81,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "docs", "docs", "{2F501E14-3
 		..\documentation\NUnit2039.md = ..\documentation\NUnit2039.md
 		..\documentation\NUnit2040.md = ..\documentation\NUnit2040.md
 		..\documentation\NUnit2041.md = ..\documentation\NUnit2041.md
+		..\documentation\NUnit2043.md = ..\documentation\NUnit2043.md
```
[the remark this answers is not in the document]
Comment (manfred-brands) [reply src/nunit.analyzers.sln:84]: Done.

File: src/nunit.analyzers/Constants/ComparisonConstraintUsageConstants.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace NUnit.Analyzers.Constants
+{
+    internal static class ComparisonConstraintUsageConstants
+    {
+        internal const string Title = "Use ComparisonConstraint for better assertion messages in case of failure";
+        internal const string Message = "Use {0} constraint instead of direct comparison for better assertion messages in case of failure";
```
[the remark this answers is not in the document]
Comment (manfred-brands) [reply src/nunit.analyzers/Constants/ComparisonConstraintUsageConstants.cs:6]: That would be fine if we would also change the 30 other rule constants Message fields that contain a '{' and are not called ...Format. 
Personally I'm not in favour of these 'Constants' classes. a) They are not with the rule and b) Only used in the rule. 
If they are only used once, I would prefer drop these classes or at least move them in the same directory as the rule.
But we are getting off-topic for this PR.

[the remark this answers is not in the document]
Comment (mikkelbu) [reply src/nunit.analyzers/Constants/ComparisonConstraintUsageConstants.cs:6]: My mistake, I thought we usually called them `MessageFormat`, but I can see that we only do this in one place. I think it make sense to move them closer to the rule, but let's do this in another PR.

**Comment by manfred-brands** [comment]
> EDIT: I guess you based your branch on `DiagnosticSuppressor`, so currently this PR also contains all the changes from #325, so I only reviewed the new files.

Yes, I created the new branch, but forgot to rebase it on master as the time.

**Comment by manfred-brands** [comment]
> So if one of the sides of the binary operator is a constant then we know that this side is the expected value (and then this should be placed inside the expected expression. If neither side is a constant, then I think we could keep the current translation as fallback.

Done. Because you asked to nicely.
Offcourse if they had installed style cop it would have told them not to do these silly things: _SA1131: Constant values should appear on the right-hand side of comparisons_

**Review by mikkelbu — APPROVED** [review]
Looks great @manfred-brands. I'm happy to merge this if you are 👍

**Comment by manfred-brands** [comment]
Collapsed commits, ready for merge.

**Comment by mikkelbu** [comment]
Merging it now. Thanks for the work @manfred-brands.

**Pull request merged**